examples/seq2seq/utils_seq2seq.py

```diff
@@ -0,0 +1,226 @@
+        tokenizer.cls_token,
+        tokenizer.sep_token,
+        tokenizer.convert_tokens_to_ids([tokenizer.pad_token])[0],
+    )
+    sequence_a_segment_id = 0
+    mask_padding_with_zero = True
+
+    for (ex_index, example) in enumerate(examples):
+        if ex_index % 10000 == 0:
+            logger.info("Writing example %d of %d", ex_index, len(examples))
+
+        tokens = tokenizer.tokenize(example.words)
+        formatted_tokens = tokenizer.tokenize(example.formatted_words)
+
+        # Account for [CLS] and [SEP] with "- 2" and with "- 3" for RoBERTa.
+        special_tokens_count = 3 if model_type in ["roberta"] else (2 if model_type != "gpt2" else 0)
+
+        if len(tokens) > max_seq_length - special_tokens_count:
+            tokens = tokens[: (max_seq_length - special_tokens_count)]
+            formatted_tokens = formatted_tokens[: (max_seq_length - special_tokens_count)]
+
+        # The convention in BERT is:
+        # (a) For sequence pairs:
+        #  tokens:   [CLS] is this jack ##son ##ville ? [SEP] no it is not . [SEP]
+        #  type_ids:   0   0  0    0    0     0       0   0   1  1  1  1   1   1
+        # (b) For single sequences:
+        #  tokens:   [CLS] the dog is hairy . [SEP]
+        #  type_ids:   0   0   0   0  0     0   0
+        #
+        # Where "type_ids" are used to indicate whether this is the first
+        # sequence or the second sequence. The embedding vectors for `type=0` and
+        # `type=1` were learned during pre-training and are added to the wordpiece
+        # embedding vector (and position vector). This is not *strictly* necessary
+        # since the [SEP] token unambiguously separates the sequences, but it makes
+        # it easier for the model to learn the concept of sequences.
+        #
+        # For classification tasks, the first vector (corresponding to [CLS]) is
+        # used as as the "sentence vector". Note that this only makes sense because
+        # the entire model is fine-tuned.
+        if model_type not in ["gpt2"]:
+            tokens += [sep_token]
+            formatted_tokens += [pad_token_label_id]
+
+        if model_type in ["roberta"]:
+            # roberta uses an extra separator b/w pairs of sentences, cf. github.com/pytorch/fairseq/commit/1684e166e3da03f5b600dbb7855cb98ddfcd0805
+            tokens += [sep_token]
+            formatted_tokens += [pad_token_label_id]
+        segment_ids = [sequence_a_segment_id] * len(tokens)
+
+        # defines the location of the CLS token:
+        # - False (Default, BERT/XLM pattern): [CLS] + A + [SEP] + B + [SEP]
+        # - True (XLNet/GPT pattern): A + [SEP] + B + [SEP] + [CLS]
+        # xlnet/gpt is special
+        if model_type in ["xlnet"]:
+            tokens += [cls_token]
+            formatted_tokens += [pad_token_label_id]
+            segment_ids += [cls_token_segment_id]
+        # gpt2 has no cls_token
+        elif model_type not in ["gpt2"]:
```

Comment:
I believe this is supposed to be 

elif model_type in ['gpt2']

 instead of "not in"